airbyte-server/src/main/java/io/airbyte/server/handlers/JobHistoryHandler.java

```diff
@@ -46,16 +51,29 @@ public JobHistoryHandler(JobPersistence jobPersistence) {
     this.jobPersistence = jobPersistence;
   }
 
+  @SuppressWarnings("UnstableApiUsage")
   public JobReadList listJobsFor(JobListRequestBody request) throws IOException {
-    final JobConfig.ConfigType configType = Enums.convertTo(request.getConfigType(), JobConfig.ConfigType.class);
-    final String configId = request.getConfigId();
+    Preconditions.checkNotNull(request.getConfigTypes(), "configType cannot be null.");
+    Preconditions.checkState(!request.getConfigTypes().isEmpty(), "Must include at least one configType.");
 
-    final List<JobWithAttemptsRead> jobReads = jobPersistence.listJobs(configType, configId)
+    final List<JobConfig.ConfigType> configTypes = request.getConfigTypes()
         .stream()
-        .map(JobConverter::getJobWithAttemptsRead)
+        .map(type -> Enums.convertTo(type, JobConfig.ConfigType.class))
         .collect(Collectors.toList());
+    final String configId = request.getConfigId();
+
+    // get jobs for each type and merge them into a single list sorted by created at.
+    Iterable<JobWithAttemptsRead> jobReads = ImmutableList.of();
+    for (final JobConfig.ConfigType configType : configTypes) {
+      final List<JobWithAttemptsRead> jobReadsForType = jobPersistence.listJobs(configType, configId)
+          .stream()
+          .map(JobConverter::getJobWithAttemptsRead)
+          .collect(Collectors.toList());
+
+      jobReads = Iterables.mergeSorted(ImmutableList.of(jobReads, jobReadsForType), Comparator.comparing(v -> v.getJob().getCreatedAt()));
+    }
 
-    return new JobReadList().jobs(jobReads);
+    return new JobReadList().jobs(StreamSupport.stream(jobReads.spliterator(), false).collect(Collectors.toList()));
```

Comment:
Can we put this logic in a common helper (`MoreStreams`?)? I believe I've seen it elsewhere in the code.

Reply:
hahah sure. this is the syntactic sugar version according to google 😉